ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -387,8 +460,15 @@ private void updateStateOfReplicas(List<InstanceConfig> instanceConfigs) {
                 Set<String> sealedReplicas = new HashSet<>(getSealedReplicas(instanceConfig));
                 Set<String> stoppedReplicas = new HashSet<>(getStoppedReplicas(instanceConfig));
                 for (AmbryReplica replica : ambryDataNodeToAmbryReplicas.get(node)) {
-                  replica.setSealedState(sealedReplicas.contains(replica.getPartitionId().toPathString()));
-                  replica.setStoppedState(stoppedReplicas.contains(replica.getPartitionId().toPathString()));
+                  if (clusterMapConfig.clusterMapEnablePartitionOverride && partitionOverrideInfoMap.containsKey(
+                      replica.getPartitionId().toPathString())) {
```

Comment:
minor nit: construct this path string once and save it to a local variable

Reply:
Fixed. Thanks.